### PR TITLE
Ensure the `-scaled` image remains in the original uploaded format

### LINKED ISSUE #67: Image mime type is not set correctly when a `webp` image is generated.
## Note

Make sure the `Performance` plugin is enabled and activated, as well with the `webp` modules for images at: `Settings > 
Performance`

Related with issue: 

- #16

## Steps to recreate:

1. Open the WordPress dashboard.
2. Go to `Media > Add New`
2. Upload the image
3. Go to `Media > Library` and open the image, observe the image has an extension `image/jpeg` instead of `image/webp`

## Original image to replicate the issue.

gantas-vaiciulenas-ZLMHrnSyJpM-unsplash

### PR DESCRIPTION
**Pull request by adamsilverstein**
## Summary

When uploading very large images (over the `big_image_size_threshold` filter threshold), WordPress creates a scaled version of the image with the word`-scaled` appended to the file name. It stores a reference to the original image in the image meta `original_image` field, see [this make post](https://make.wordpress.org/core/2019/10/09/introducing-handling-of-big-images-in-wordpress-5-3/#:~:text=The%20default%20threshold%20value%20is,as%20the%20largest%20available%20size.) for more details.

WordPress subsequently uses the "scaled" image as if it were the original in most use cases (only going back to the true original image when editing the image). This causes the mime type shown in the media library for a large jpeg upload to show as image/webp because the scaled image is a WebP.

To fix that, in this PR I skip the `-scaled` image, leaving it in the original format of the uploaded image.

This matches the behavior of uploading a smaller image that isn't over the `big_image_size_threshold`, improving consistency.

Fixes https://github.com/WordPress/performance/issues/67

## Relevant technical choices

* Makes sense that the `scaled` image, which WordPress largely treats as the "original" would remain in the mime type of the upload
* Ensure the behavior is consistent between large and smaller image uploads.
 
## Checklist

- [x] PR has either `[Focus]` or `Infrastructure` label.
- [x] PR has a `[Type]` label.
- [x] PR has a milestone or the `no milestone` label.

### CODE REVIEW

File: modules/images/webp-uploads/load.php [hunk 1]
```diff
@@ -31,6 +31,11 @@ function webp_uploads_filter_image_editor_output_format( $output_format, $filena
 		return $output_format;
 	}
 
+	// Skip conversion when creating the `-scaled` image (for large image uploads).
+	if ( 1 === preg_match( '/.*-scaled\..{3}.?$/', $filename ) ) {
+		return $output_format;
+	}
```
Comment (felixarntz) [inline modules/images/webp-uploads/load.php:37]: Can we add a test for this, with a couple different file names?

File: modules/images/webp-uploads/load.php [hunk 2]
```diff
@@ -31,6 +31,11 @@ function webp_uploads_filter_image_editor_output_format( $output_format, $filena
 		return $output_format;
 	}
 
+	// Skip conversion when creating the `-scaled` image (for large image uploads).
+	if ( 1 === preg_match( '/.*-scaled\..{3}.?$/', $filename ) ) {
```
Comment (felixarntz) [inline modules/images/webp-uploads/load.php]: The `1 === ` is maybe a bit overly strict, we can probably remove it. I'm not sure, but I'm thinking maybe some obscure PHP version returns an integer here, another a boolean - in any case, for simple boolean(-ish) checks I don't think we need to have strict checks, similar to how it's done in other such places in WP core.

**Review by felixarntz — COMMENTED** [review]
Looks good, just two comments.

File: modules/images/webp-uploads/load.php (hunk 2 above)
Comment (mitogh) [inline modules/images/webp-uploads/load.php]: Similar comment to @felixarntz would be ideal to check `!== false`  instead.

Additionally, the first part of the REGEX can be removed.

```suggestion
	if ( false !== preg_match( '/-scaled\..{3}.?$/', $filename ) ) {
```

Another approach as well can be using substr.

```php
if ( substr( $filename, -11, 7 ) === '-scaled' ) {
```

**Comment by mitogh** [comment]
Additionally if we are going with this route once this is merged we should be able to close: 

- #122

File: modules/images/webp-uploads/load.php (hunk 2 above)
Comment (adamsilverstein) [reply modules/images/webp-uploads/load.php]: i wanted to avoid substring which might make it harder to match files with .jpeg vs .jpg

Comment (adamsilverstein) [reply modules/images/webp-uploads/load.php]: 👍🏼  updated

Comment (adamsilverstein) [reply modules/images/webp-uploads/load.php]: updated regex. return of false is a regex failure, 0 not found, 1 found, so i changed to a "truthy" check, i'll make sure that works

**Comment by adamsilverstein** [comment]
> Additionally if we are going with this route once this is merged we should be able to close:
> 
> * [`*.jpg` is not generated for scaled full size image. #122](https://github.com/WordPress/performance/issues/122)

Thinking we should generate both, see https://github.com/WordPress/performance/issues/142#issuecomment-1028354600

File: modules/images/webp-uploads/load.php (hunk 1 above)
Comment (adamsilverstein) [reply modules/images/webp-uploads/load.php:37]: Added in [`98e1003` (#143)](https://github.com/WordPress/performance/pull/143/commits/98e100336738c0fd4b1ad8c56536b10d1a3825fc)

File: tests/modules/images/webp-uploads/webp-uploads-test.php [hunk 3]
```diff
@@ -3,9 +3,12 @@
  * Tests for webp-uploads module.
  *
  * @package performance-lab
+ * @group webp-uploads
```
Comment (adamsilverstein) [inline tests/modules/images/webp-uploads/webp-uploads-test.php:6]: Adding this enables running just these tests locally

Comment (felixarntz) [reply tests/modules/images/webp-uploads/webp-uploads-test.php:6]: Great! Maybe worth adding such an annotation to the other module test class(es) as well? Doesn't really belong here, but might make sense to just add.

File: tests/modules/images/webp-uploads/webp-uploads-test.php [hunk 4]
```diff
@@ -3,9 +3,12 @@
...
  */
 
 class WebP_Uploads_Tests extends WP_UnitTestCase {
+	function return_webp_array() {
+		return array( 'image/webp' ); }
```
Comment (felixarntz) [inline tests/modules/images/webp-uploads/webp-uploads-test.php]: The syntax here looks off? Can you move the closing brace below?

File: tests/modules/images/webp-uploads/webp-uploads-test.php [hunk 5]
```diff
@@ -17,7 +20,9 @@ function test_webp_uploads_applies_filter_based_on_system_support() {
 		} else {
 			$expect = array( 'image/jpeg' => 'image/webp' );
 		}
-
+		// The image_editor_output_format filter is only applied if image_editor_output_format
+		// has fewer than two types.
+		add_filter( 'image_editor_output_formats', array( $this, 'return_webp_array' ) );
```
Comment (felixarntz) [inline tests/modules/images/webp-uploads/webp-uploads-test.php]: Why did you add this here? This existing test shouldn't be affected by this change, or is it?

File: tests/modules/images/webp-uploads/webp-uploads-test.php [hunk 6]
```diff
@@ -28,6 +33,7 @@ function test_webp_uploads_applies_filter_based_on_system_support() {
 	function test_webp_uploads_filter_image_editor_output_format_with_support() {
 		// Mock a system that supports WebP.
 		add_filter( 'wp_image_editors', array( $this, 'mock_wp_image_editor_supports' ) );
+		add_filter( 'image_editor_output_formats', array( $this, 'return_webp_array' ) );
```
Comment (felixarntz) [inline tests/modules/images/webp-uploads/webp-uploads-test.php]: See above, why does this need to be here?

File: tests/modules/images/webp-uploads/webp-uploads-test.php (hunk 4 above)
Comment (felixarntz) [reply tests/modules/images/webp-uploads/webp-uploads-test.php]: Ah I see you changed it now, nevermind.

Comment (adamsilverstein) [reply tests/modules/images/webp-uploads/webp-uploads-test.php]: oof phpcbf did that, i was shooting for a single line function which phpcs didn't like; fixed.

File: tests/modules/images/webp-uploads/webp-uploads-test.php (hunk 5 above)
Comment (adamsilverstein) [reply tests/modules/images/webp-uploads/webp-uploads-test.php]: oh, you are right, I thought this was breaking the tests, removing.

File: tests/modules/images/webp-uploads/webp-uploads-test.php (hunk 6 above)
Comment (adamsilverstein) [reply tests/modules/images/webp-uploads/webp-uploads-test.php]: removing
